From 58cb415a2aca97609ce51462e9ea42e65183b0b4 Mon Sep 17 00:00:00 2001
From: Pengpeng Cui <pengpengx.cui@intel.com>
Date: Mon, 2 Feb 2015 19:15:53 +0800
Subject: [PATCH 254/255] atomisp2: css2400 for irci_master_20150131_0457

Merge CSS2.1 for css2400  for
irci_master_20150131_0457 firmware

Change-Id: Iee91c0f230130e74dc8c568c2c4e0743fe98f8f7
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-20062
Signed-off-by: Pengpeng Cui <pengpengx.cui@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/38923
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/css2400b0_v21/css_api_version.h   | 18 ++++++---
 .../media/pci/atomisp2/css2400b0_v21/ia_css.h      |  4 +-
 .../pci/atomisp2/css2400b0_v21/ia_css_host_data.h  |  4 +-
 .../pci/atomisp2/css2400b0_v21/ia_css_types.h      |  4 +-
 .../atomisp2/css2400b0_v21/ia_css_version_data.h   |  2 +-
 .../isp/kernels/eed1_8/ia_css_eed1_8_param.h       | 29 +++++++-------
 .../isp/kernels/hdr/ia_css_hdr.host.c              |  4 +-
 .../isp/kernels/hdr/ia_css_hdr.host.h              |  4 +-
 .../isp/kernels/hdr/ia_css_hdr_param.h             |  4 +-
 .../isp/kernels/hdr/ia_css_hdr_types.h             |  4 +-
 .../io_ls/bayer_io_ls/ia_css_bayer_io_param.h      | 11 +-----
 .../io_ls/bayer_io_ls/ia_css_bayer_io_types.h      | 12 +-----
 .../kernels/io_ls/common/ia_css_common_io_param.h  | 30 +++++++++++++++
 .../kernels/io_ls/common/ia_css_common_io_types.h  | 39 +++++++++++++++++++
 .../io_ls/plane_io_ls/ia_css_plane_io_param.h      | 11 +-----
 .../io_ls/plane_io_ls/ia_css_plane_io_types.h      | 16 ++------
 .../io_ls/yuv420_io_ls/ia_css_yuv420_io_param.h    | 12 +-----
 .../io_ls/yuv420_io_ls/ia_css_yuv420_io_types.h    | 12 +-----
 .../io_ls/yuv444_io_ls/ia_css_yuv444_io_param.h    | 20 ++++++++++
 .../io_ls/yuv444_io_ls/ia_css_yuv444_io_types.h    | 20 ++++++++++
 .../isp/kernels/s3a/s3a_1.0/ia_css_s3a_types.h     |  6 +--
 .../isp/modes/interface/input_buf.isp.h            |  1 +
 .../runtime/binary/interface/ia_css_binary.h       |  2 +-
 .../css2400b0_v21/runtime/binary/src/binary.c      | 12 ++++--
 .../runtime/bufq/interface/ia_css_bufq_comm.h      |  8 +++-
 .../media/pci/atomisp2/css2400b0_v21/sh_css.c      | 19 ++++++++--
 .../pci/atomisp2/css2400b0_v21/sh_css_firmware.c   |  4 +-
 .../pci/atomisp2/css2400b0_v21/sh_css_params.c     |  5 ++-
 .../media/pci/atomisp2/css2400b0_v21/spmem_dump.c  | 44 +++++++++++-----------
 29 files changed, 220 insertions(+), 141 deletions(-)
 create mode 100644 drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/common/ia_css_common_io_param.h
 create mode 100644 drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/common/ia_css_common_io_types.h
 create mode 100644 drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv444_io_ls/ia_css_yuv444_io_param.h
 create mode 100644 drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv444_io_ls/ia_css_yuv444_io_types.h

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/css_api_version.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/css_api_version.h
index cb7a000..303fdf9 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/css_api_version.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/css_api_version.h
@@ -35,7 +35,7 @@ The version string has four dot-separated numbers, read left to right:
 
 */
 
-#define CSS_API_VERSION_STRING	"2.1.10.6"
+#define CSS_API_VERSION_STRING	"2.1.11.2"
 
 /*
 Change log
@@ -235,10 +235,6 @@ v2.1.10.4 (4 changes parallel), fix IEFD's puclic API types
 v2.1.10.4 (4 changes parallel), Add IA_CSS_FW_WARNING_FRAME_PARAM_MISMATCH
 - Add IA_CSS_FW_WARNING_FRAME_PARAM_MISMATCH enum to ia_css_fw_warning type
 
-v2.1.10.5 (5 changes parallel), add_flag_to_disable_continous_viewfinder
-- add a new flag in stream_config to disable continuous viewfinder
-- in ZSL use case.
-
 v2.1.10.5 (5 changes parallel), change the pipe version type from integer to enum
 - add new enum to enumerate ISP pipe version
 - change the pipe version type in pipe_config from integer to enum
@@ -246,7 +242,17 @@ v2.1.10.5 (5 changes parallel), change the pipe version type from integer to enu
 v2.1.10.5 (2 changes parallel), Added public interface for setting the scaler LUT.
 - Added the public struct to output system and modified the global config struct.
 
-v2.1.10.6, new API getter functions for gdc in buffer information
+v2.1.10.6 (6 changes parallel), add_flag_to_disable_continous_viewfinder
+- add a new flag in stream_config to disable continuous viewfinder
+- in ZSL use case.
+
+v2.1.11.0 (2 changes parallel), Stop Support for Skycam B0
+- Remove a few pre-processor defines for Skycam B0/C0 as support
+
+v2.1.11.2 (3 changes parallel), Enlarge AF AWB_FR stats buffers
+- Enlarge AF and AWB_FR stats buffers to support max grid width per stripe as oppose to per frame
+
+v2.1.11.2 (2 changes parallel), new API getter functions for gdc in buffer information
 - ia_css_pipe_get_dvs_filter() added
 - ia_css_pipe_get_gdc_in_buffer_info() added
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css.h
index 8021eedf..712964d 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css.h
@@ -1,5 +1,5 @@
-/* Release Version: irci_master_20150129_1500 */
-/* Release Version: irci_master_20150129_1500 */
+/* Release Version: irci_master_20150131_0457 */
+/* Release Version: irci_master_20150131_0457 */
 /*
  * Support for Intel Camera Imaging ISP subsystem.
  * Copyright (c) 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_host_data.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_host_data.h
index 022b232..d28e15d 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_host_data.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_host_data.h
@@ -1,5 +1,5 @@
-/* Release Version: irci_master_20150129_1500 */
-/* Release Version: irci_master_20150129_1500 */
+/* Release Version: irci_master_20150131_0457 */
+/* Release Version: irci_master_20150131_0457 */
 /*
  * Support for Intel Camera Imaging ISP subsystem.
  * Copyright (c) 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_types.h
index 8e43819..a7446a8 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_types.h
@@ -1,5 +1,5 @@
-/* Release Version: irci_master_20150129_1500 */
-/* Release Version: irci_master_20150129_1500 */
+/* Release Version: irci_master_20150131_0457 */
+/* Release Version: irci_master_20150131_0457 */
 /*
  * Support for Intel Camera Imaging ISP subsystem.
  * Copyright (c) 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_version_data.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_version_data.h
index 7b2c3c4..7c48c58 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_version_data.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/ia_css_version_data.h
@@ -22,7 +22,7 @@
 #define __IA_CSS_VERSION_DATA_H
 
 
-#define CSS_VERSION_STRING "REL:20150129_05.4_1637; API:2.1.10.6; GIT:scci_20150129_0104_#04ffe549c0c12783fedda9e5ff6703a6a446a50c; SDK:/nfs/iir/disks/iir_hivepackages_003/iir_hivepkgs_disk017/Css_Mizuchi/packages/Css_Mizuchi/int_css_mizuchi_20140829_1053; USER:viedifw; "
+#define CSS_VERSION_STRING "REL:20150131_05.6_0526; API:2.1.11.2; GIT:irci_20150130_1500__5ef052#5ef052995cc2fc8b1d29fa5ed307c5f5bb2d73a7; SDK:/nfs/iir/disks/iir_hivepackages_003/iir_hivepkgs_disk017/Css_Mizuchi/packages/Css_Mizuchi/int_css_mizuchi_20140829_1053; USER:viedifw; "
 
 
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/eed1_8/ia_css_eed1_8_param.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/eed1_8/ia_css_eed1_8_param.h
index 32cbfb5..0260e6c 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/eed1_8/ia_css_eed1_8_param.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/eed1_8/ia_css_eed1_8_param.h
@@ -32,12 +32,9 @@
 
 /* Coring Threshold minima
  * Used in Tint color suppression.
- * Default: ??
- */
-/* TODO: Find propper value for this parameter.
- * KFS currently does not specify this.
+ * Default: 1
  */
-#define EED1_8_CORINGTHMIN	1024
+#define EED1_8_CORINGTHMIN	1
 
 /* Define size of the state..... TODO: check if this is the correct place */
 /* 4 planes : GR, R, B, GB */
@@ -55,24 +52,24 @@
 #define EED1_8_STATE_LD_H_HEIGHT	(2 * NUM_PLANES)
 #define EED1_8_STATE_LD_H_WIDTH		CEIL_DIV(MAX_FRAME_SIMDWIDTH, 2)
 
-/* 2 lines stater per color plane LD_V state */
+/* 2 lines state per color plane LD_V state */
 #define EED1_8_STATE_LD_V_HEIGHT	(2 * NUM_PLANES)
 #define EED1_8_STATE_LD_V_WIDTH		CEIL_DIV(MAX_FRAME_SIMDWIDTH, 2)
 
-/* 2 lines (single plane) state for D_Hr state */
+/* 1 line (single plane) state for D_Hr state */
 #define EED1_8_STATE_D_HR_HEIGHT	1
 #define EED1_8_STATE_D_HR_WIDTH		CEIL_DIV(MAX_FRAME_SIMDWIDTH, 2)
 
-/* 3 lines (single plane) state for D_Hb state */
-#define EED1_8_STATE_D_HB_HEIGHT	2
+/* 1 line (single plane) state for D_Hb state */
+#define EED1_8_STATE_D_HB_HEIGHT	1
 #define EED1_8_STATE_D_HB_WIDTH		CEIL_DIV(MAX_FRAME_SIMDWIDTH, 2)
 
-/* 3 lines (single plane) state for D_Vr state */
+/* 2 lines (single plane) state for D_Vr state */
 #define EED1_8_STATE_D_VR_HEIGHT	2
 #define EED1_8_STATE_D_VR_WIDTH		CEIL_DIV(MAX_FRAME_SIMDWIDTH, 2)
 
-/* 2 lines (single plane) state for D_Vb state */
-#define EED1_8_STATE_D_VB_HEIGHT	1
+/* 2 line (single plane) state for D_Vb state */
+#define EED1_8_STATE_D_VB_HEIGHT	2
 #define EED1_8_STATE_D_VB_WIDTH		CEIL_DIV(MAX_FRAME_SIMDWIDTH, 2)
 
 /* 3 lines state for R and B (= 2 planes) rb_zipped_state */
@@ -84,12 +81,12 @@
 #define EED1_8_STATE_YC_HEIGHT	1
 #define EED1_8_STATE_YC_WIDTH	MAX_FRAME_SIMDWIDTH
 
-/* 2 lines state per color plane Cg_state */
-#define EED1_8_STATE_CG_HEIGHT	(2 * NUM_PLANES)
+/* 1 line state per color plane Cg_state */
+#define EED1_8_STATE_CG_HEIGHT	(1 * NUM_PLANES)
 #define EED1_8_STATE_CG_WIDTH	CEIL_DIV(MAX_FRAME_SIMDWIDTH, 2)
 
-/* 2 lines state per color plane Co_state */
-#define EED1_8_STATE_CO_HEIGHT	(2 * NUM_PLANES)
+/* 1 line state per color plane Co_state */
+#define EED1_8_STATE_CO_HEIGHT	(1 * NUM_PLANES)
 #define EED1_8_STATE_CO_WIDTH	CEIL_DIV(MAX_FRAME_SIMDWIDTH, 2)
 
 /* 1 full input line (GR-R color line) for AbsK state */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr.host.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr.host.c
index 2e5a909..e2e828b 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr.host.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr.host.c
@@ -1,5 +1,5 @@
-/* Release Version: irci_master_20150129_1500 */
-/* Release Version: irci_master_20150129_1500 */
+/* Release Version: irci_master_20150131_0457 */
+/* Release Version: irci_master_20150131_0457 */
 /*
  * INTEL CONFIDENTIAL
  *
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr.host.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr.host.h
index 73ed930..d2197b1 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr.host.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr.host.h
@@ -1,5 +1,5 @@
-/* Release Version: irci_master_20150129_1500 */
-/* Release Version: irci_master_20150129_1500 */
+/* Release Version: irci_master_20150131_0457 */
+/* Release Version: irci_master_20150131_0457 */
 /*
  * Support for Intel Camera Imaging ISP subsystem.
  * Copyright (c) 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr_param.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr_param.h
index 58dc99c..b55ff4b 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr_param.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr_param.h
@@ -1,5 +1,5 @@
-/* Release Version: irci_master_20150129_1500 */
-/* Release Version: irci_master_20150129_1500 */
+/* Release Version: irci_master_20150131_0457 */
+/* Release Version: irci_master_20150131_0457 */
 /*
  * Support for Intel Camera Imaging ISP subsystem.
  * Copyright (c) 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr_types.h
index 65c4d81..a8fca14 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/hdr/ia_css_hdr_types.h
@@ -1,5 +1,5 @@
-/* Release Version: irci_master_20150129_1500 */
-/* Release Version: irci_master_20150129_1500 */
+/* Release Version: irci_master_20150131_0457 */
+/* Release Version: irci_master_20150131_0457 */
 /*
  * Support for Intel Camera Imaging ISP subsystem.
  * Copyright (c) 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/bayer_io_ls/ia_css_bayer_io_param.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/bayer_io_ls/ia_css_bayer_io_param.h
index ecf86bd..9fe58b2 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/bayer_io_ls/ia_css_bayer_io_param.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/bayer_io_ls/ia_css_bayer_io_param.h
@@ -15,15 +15,6 @@
 #ifndef __IA_CSS_BAYER_IO_PARAM
 #define __IA_CSS_BAYER_IO_PARAM
 
-#include "ia_css_bayer_io_types.h"
-
-struct bayer_io_configuration {
-	unsigned base_address;
-	unsigned width;
-	unsigned height;
-	unsigned stride;
-	unsigned ddr_elems_per_word;
-	unsigned dma_channel[NUM_BAYER_DMA_CHANNELS];
-};
+#include "../common/ia_css_common_io_param.h"
 
 #endif /* __IA_CSS_BAYER_IO_PARAM */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/bayer_io_ls/ia_css_bayer_io_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/bayer_io_ls/ia_css_bayer_io_types.h
index 77e1787..c2a8336 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/bayer_io_ls/ia_css_bayer_io_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/bayer_io_ls/ia_css_bayer_io_types.h
@@ -12,19 +12,9 @@
  * more details.
  */
 
-
 #ifndef __IA_CSS_BAYER_IO_TYPES_H
 #define __IA_CSS_BAYER_IO_TYPES_H
 
-#define NUM_BAYER_DMA_CHANNELS 1
-
-struct ia_css_bayer_io_config {
-	unsigned base_address;
-	unsigned width;
-	unsigned height;
-	unsigned stride;
-	unsigned ddr_elems_per_word;
-	unsigned dma_channel[NUM_BAYER_DMA_CHANNELS];
-};
+#include "../common/ia_css_common_io_types.h"
 
 #endif /* __IA_CSS_BAYER_IO_TYPES_H */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/common/ia_css_common_io_param.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/common/ia_css_common_io_param.h
new file mode 100644
index 0000000..24b2545
--- /dev/null
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/common/ia_css_common_io_param.h
@@ -0,0 +1,30 @@
+/*
+ * INTEL CONFIDENTIAL
+ *
+ * Copyright (C) 2010 - 2015 Intel Corporation.
+ * All Rights Reserved.
+ *
+ * The source code contained or described herein and all documents
+ * related to the source code ("Material") are owned by Intel Corporation
+ * or licensors. Title to the Material remains with Intel
+ * Corporation or its licensors. The Material contains trade
+ * secrets and proprietary and confidential information of Intel or its
+ * licensors. The Material is protected by worldwide copyright
+ * and trade secret laws and treaty provisions. No part of the Material may
+ * be used, copied, reproduced, modified, published, uploaded, posted,
+ * transmitted, distributed, or disclosed in any way without Intel's prior
+ * express written permission.
+ *
+ * No License under any patent, copyright, trade secret or other intellectual
+ * property right is granted to or conferred upon you by disclosure or
+ * delivery of the Materials, either expressly, by implication, inducement,
+ * estoppel or otherwise. Any license under such intellectual property rights
+ * must be express and approved by Intel in writing.
+ */
+
+#ifndef __IA_CSS_COMMON_IO_PARAM
+#define __IA_CSS_COMMON_IO_PARAM
+
+#include "../common/ia_css_common_io_types.h"
+
+#endif /* __IA_CSS_COMMON_IO_PARAM */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/common/ia_css_common_io_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/common/ia_css_common_io_types.h
new file mode 100644
index 0000000..e9af0e6
--- /dev/null
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/common/ia_css_common_io_types.h
@@ -0,0 +1,39 @@
+/*
+ * INTEL CONFIDENTIAL
+ *
+ * Copyright (C) 2010 - 2015 Intel Corporation.
+ * All Rights Reserved.
+ *
+ * The source code contained or described herein and all documents
+ * related to the source code ("Material") are owned by Intel Corporation
+ * or licensors. Title to the Material remains with Intel
+ * Corporation or its licensors. The Material contains trade
+ * secrets and proprietary and confidential information of Intel or its
+ * licensors. The Material is protected by worldwide copyright
+ * and trade secret laws and treaty provisions. No part of the Material may
+ * be used, copied, reproduced, modified, published, uploaded, posted,
+ * transmitted, distributed, or disclosed in any way without Intel's prior
+ * express written permission.
+ *
+ * No License under any patent, copyright, trade secret or other intellectual
+ * property right is granted to or conferred upon you by disclosure or
+ * delivery of the Materials, either expressly, by implication, inducement,
+ * estoppel or otherwise. Any license under such intellectual property rights
+ * must be express and approved by Intel in writing.
+ */
+
+#ifndef __IA_CSS_COMMON_IO_TYPES
+#define __IA_CSS_COMMON_IO_TYPES
+
+#define MAX_IO_DMA_CHANNELS 2
+
+struct ia_css_common_io_config {
+	unsigned base_address;
+	unsigned width;
+	unsigned height;
+	unsigned stride;
+	unsigned ddr_elems_per_word;
+	unsigned dma_channel[MAX_IO_DMA_CHANNELS];
+};
+
+#endif /* __IA_CSS_COMMON_IO_TYPES */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/plane_io_ls/ia_css_plane_io_param.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/plane_io_ls/ia_css_plane_io_param.h
index dce2b0b..9300568 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/plane_io_ls/ia_css_plane_io_param.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/plane_io_ls/ia_css_plane_io_param.h
@@ -25,15 +25,6 @@
 #ifndef __IA_CSS_PLANE_IO_PARAM_H
 #define __IA_CSS_PLANE_IO_PARAM_H
 
-#include "ia_css_plane_io_types.h"
-
-struct sh_css_plane_io_config {
-	unsigned base_address;
-	unsigned width;
-	unsigned height;
-	unsigned stride;
-	unsigned ddr_elems_per_word;
-	unsigned dma_channel[PLANE_IO_LS_NUM_DMA_CHANNELS];
-};
+#include "../common/ia_css_common_io_param.h"
 
 #endif /* __IA_CSS_PLANE_IO_PARAM_H */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/plane_io_ls/ia_css_plane_io_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/plane_io_ls/ia_css_plane_io_types.h
index c11b540..57ed18f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/plane_io_ls/ia_css_plane_io_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/plane_io_ls/ia_css_plane_io_types.h
@@ -25,21 +25,13 @@
 #ifndef __IA_CSS_PLANE_IO_TYPES_H
 #define __IA_CSS_PLANE_IO_TYPES_H
 
-#define PLANE_IO_LS_NUM_PLANES       3
-#define PLANE_IO_LS_NUM_DMA_CHANNELS 1
+#include "../common/ia_css_common_io_types.h"
 
-struct plane_io_config {
-	unsigned base_address;
-	unsigned width;
-	unsigned height;
-	unsigned stride;
-	unsigned ddr_elems_per_word;
-	unsigned dma_channel[PLANE_IO_LS_NUM_DMA_CHANNELS];
-};
+#define PLANE_IO_LS_NUM_PLANES       3
 
 struct ia_css_plane_io_config {
-	struct plane_io_config get_plane_io_config[PLANE_IO_LS_NUM_PLANES];
-	struct plane_io_config put_plane_io_config[PLANE_IO_LS_NUM_PLANES];
+	struct ia_css_common_io_config get_plane_io_config[PLANE_IO_LS_NUM_PLANES];
+	struct ia_css_common_io_config put_plane_io_config[PLANE_IO_LS_NUM_PLANES];
 };
 
 #endif /* __IA_CSS_PLANE_IO_TYPES_H */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv420_io_ls/ia_css_yuv420_io_param.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv420_io_ls/ia_css_yuv420_io_param.h
index 5453a19..cf27054 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv420_io_ls/ia_css_yuv420_io_param.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv420_io_ls/ia_css_yuv420_io_param.h
@@ -15,16 +15,6 @@
 #ifndef __IA_CSS_YUV420_IO_PARAM
 #define __IA_CSS_YUV420_IO_PARAM
 
-#define NUM_YUV420_DMA_CHANNELS 2
-
-struct ia_css_yuv420_io_params {
-	unsigned base_address;
-	unsigned width;
-	unsigned height;
-	unsigned stride;
-	unsigned ddr_elems_per_word;
-	unsigned dma_channel[NUM_YUV420_DMA_CHANNELS];
-};
-
+#include "../common/ia_css_common_io_param.h"
 
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv420_io_ls/ia_css_yuv420_io_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv420_io_ls/ia_css_yuv420_io_types.h
index 5c01e3a..ffb9413 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv420_io_ls/ia_css_yuv420_io_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv420_io_ls/ia_css_yuv420_io_types.h
@@ -15,16 +15,6 @@
 #ifndef __IA_CSS_YUV420_IO_TYPES
 #define __IA_CSS_YUV420_IO_TYPES
 
-#define NUM_YUV420_DMA_CHANNELS 2
-
-struct ia_css_yuv420_io_config {
-	unsigned base_address;
-	unsigned width;
-	unsigned height;
-	unsigned stride;
-	unsigned ddr_elems_per_word;
-	unsigned dma_channel[NUM_YUV420_DMA_CHANNELS];
-};
-
+#include "../common/ia_css_common_io_types.h"
 
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv444_io_ls/ia_css_yuv444_io_param.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv444_io_ls/ia_css_yuv444_io_param.h
new file mode 100644
index 0000000..6258fb4
--- /dev/null
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv444_io_ls/ia_css_yuv444_io_param.h
@@ -0,0 +1,20 @@
+/*
+ * Support for Intel Camera Imaging ISP subsystem.
+ * Copyright (c) 2015, Intel Corporation.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ */
+
+#ifndef __IA_CSS_YUV444_IO_PARAM
+#define __IA_CSS_YUV444_IO_PARAM
+
+#include "../common/ia_css_common_io_param.h"
+
+#endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv444_io_ls/ia_css_yuv444_io_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv444_io_ls/ia_css_yuv444_io_types.h
new file mode 100644
index 0000000..5d0c92a
--- /dev/null
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/io_ls/yuv444_io_ls/ia_css_yuv444_io_types.h
@@ -0,0 +1,20 @@
+/*
+ * Support for Intel Camera Imaging ISP subsystem.
+ * Copyright (c) 2015, Intel Corporation.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ */
+
+#ifndef __IA_CSS_YUV444_IO_TYPES
+#define __IA_CSS_YUV444_IO_TYPES
+
+#include "../common/ia_css_common_io_types.h"
+
+#endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/s3a/s3a_1.0/ia_css_s3a_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/s3a/s3a_1.0/ia_css_s3a_types.h
index d41d5ce..8c29e42 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/s3a/s3a_1.0/ia_css_s3a_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/kernels/s3a/s3a_1.0/ia_css_s3a_types.h
@@ -21,7 +21,7 @@
 
 #include <ia_css_frac.h>
 
-#if (defined(SYSTEM_css_skycam_a0t_system) || defined(SYSTEM_css_skycam_c0_system)) && (! defined(PIPE_GENERATION) )
+#if (defined(SYSTEM_css_skycam_c0_system)) && (! defined(PIPE_GENERATION) )
 #include "../../../../components/stats_3a/src/stats_3a_public.h"
 #endif
 
@@ -38,7 +38,7 @@
  */
 struct ia_css_3a_grid_info {
 
-#if defined(SYSTEM_css_skycam_a0t_system) || defined(SYSTEM_css_skycam_c0_system)
+#if defined(SYSTEM_css_skycam_c0_system)
 	uint32_t ae_enable;					/**< ae enabled in binary,
 								   0:disabled, 1:enabled */
 	struct ae_public_config_grid_config	ae_grd_info;	/**< see description in ae_public.h*/
@@ -97,7 +97,7 @@ struct ia_css_3a_grid_info {
 #endif
 };
 
-#if defined(SYSTEM_css_skycam_a0t_system) || defined(SYSTEM_css_skycam_c0_system)
+#if defined(SYSTEM_css_skycam_c0_system)
 #define DEFAULT_3A_GRID_INFO \
 { \
 	0,				/* ae_enable */ \
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/modes/interface/input_buf.isp.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/modes/interface/input_buf.isp.h
index a60b38e..9668350 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/modes/interface/input_buf.isp.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/isp/modes/interface/input_buf.isp.h
@@ -17,6 +17,7 @@
 
 /* Temporary include, since IA_CSS_BINARY_MODE_COPY is still needed */
 #include "sh_css_defs.h"
+#include "isp_const.h" /* MAX_VECTORS_PER_INPUT_LINE */
 
 #define INPUT_BUF_HEIGHT	2 /* double buffer */
 #define INPUT_BUF_LINES		2
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/binary/interface/ia_css_binary.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/binary/interface/ia_css_binary.h
index 37fc509..b7010f9 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/binary/interface/ia_css_binary.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/binary/interface/ia_css_binary.h
@@ -222,7 +222,7 @@ ia_css_binary_get_shading_info(const struct ia_css_binary *binary,
 			const struct ia_css_stream_config *stream_config,
 			struct ia_css_shading_info *info);
 
-void
+enum ia_css_err
 ia_css_binary_3a_grid_info(const struct ia_css_binary *binary,
 			   struct ia_css_grid_info *info,
 			   struct ia_css_pipe *pipe);
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/binary/src/binary.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/binary/src/binary.c
index a9a2147..eff89fc 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/binary/src/binary.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/binary/src/binary.c
@@ -352,14 +352,17 @@ ia_css_binary_dvs_grid_info(const struct ia_css_binary *binary,
 #endif
 }
 
-void
+enum ia_css_err
 ia_css_binary_3a_grid_info(const struct ia_css_binary *binary,
 			   struct ia_css_grid_info *info,
 			   struct ia_css_pipe *pipe)
 {
 	struct ia_css_3a_grid_info *s3a_info;
+	enum ia_css_err err = IA_CSS_SUCCESS;
+
+	IA_CSS_ENTER_PRIVATE("binary=%p, info=%p, pipe=%p",
+			     binary, info, pipe);
 
-	(void)pipe;
 	assert(binary != NULL);
 	assert(info != NULL);
 	s3a_info = &info->s3a_grid;
@@ -389,9 +392,10 @@ ia_css_binary_3a_grid_info(const struct ia_css_binary *binary,
 	s3a_info->awb_enable        = binary->info->sp.enable.awb_acc;
 	s3a_info->elem_bit_depth    = SH_CSS_BAYER_BITS;
 
-	ia_css_3a_stat_grid_calculate(s3a_info, pipe);
-
+	err = ia_css_3a_stat_grid_calculate(s3a_info, pipe);
 #endif
+	IA_CSS_LEAVE_ERR_PRIVATE(err);
+	return err;
 }
 
 static void
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/bufq/interface/ia_css_bufq_comm.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/bufq/interface/ia_css_bufq_comm.h
index 0831be8..aa845eb 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/bufq/interface/ia_css_bufq_comm.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/runtime/bufq/interface/ia_css_bufq_comm.h
@@ -33,7 +33,13 @@ enum sh_css_queue_id {
 #if defined(USE_INPUT_SYSTEM_VERSION_2) || defined(USE_INPUT_SYSTEM_VERSION_2401)
 	SH_CSS_QUEUE_H_ID, /* for metadata */
 #endif
-	SH_CSS_MAX_NUM_QUEUES
+
+#if defined(HAS_NO_INPUT_SYSTEM) || defined(USE_INPUT_SYSTEM_VERSION_2) || defined(USE_INPUT_SYSTEM_VERSION_2401)
+#define SH_CSS_MAX_NUM_QUEUES (SH_CSS_QUEUE_H_ID+1)
+#else
+#define SH_CSS_MAX_NUM_QUEUES (SH_CSS_QUEUE_G_ID+1)
+#endif
+
 };
 
 #define SH_CSS_MAX_DYNAMIC_BUFFERS_PER_THREAD SH_CSS_MAX_NUM_QUEUES
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css.c
index 2e9b18d..61caf40 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css.c
@@ -5095,13 +5095,17 @@ sh_css_pipe_get_grid_info(struct ia_css_pipe *pipe,
 
 	assert(pipe != NULL);
 	assert(info != NULL);
-	ia_css_debug_dtrace(IA_CSS_DEBUG_TRACE_PRIVATE, "sh_css_pipe_get_grid_info() enter:\n");
+
+	IA_CSS_ENTER_PRIVATE("");
 
 	binary = ia_css_pipe_get_s3a_binary(pipe);
 
-	if (binary)
-		ia_css_binary_3a_grid_info(binary, info, pipe);
-	else
+
+	if (binary) {
+		err = ia_css_binary_3a_grid_info(binary, info, pipe);
+		if (err != IA_CSS_SUCCESS)
+			goto ERR;
+	} else
 		memset(&info->s3a_grid, 0, sizeof(info->s3a_grid));
 
 	binary = ia_css_pipe_get_sdis_binary(pipe);
@@ -5125,6 +5129,8 @@ sh_css_pipe_get_grid_info(struct ia_css_pipe *pipe,
 #error "Unknown VAMEM version"
 #endif
 
+ERR:
+	IA_CSS_LEAVE_ERR_PRIVATE(err);
 	return err;
 }
 
@@ -5948,6 +5954,11 @@ allocate_delay_frames(enum ia_css_pipe_id mode,
 	struct ia_css_frame_info ref_info;
 	enum ia_css_err err = IA_CSS_SUCCESS;
 
+	if (((mode == IA_CSS_PIPE_ID_CAPTURE) && (mycs_capture == NULL)) ||
+	    ((mode == IA_CSS_PIPE_ID_VIDEO) && (mycs_video == NULL))) {
+		return IA_CSS_ERR_INVALID_ARGUMENTS;
+	 }
+
 	if (mode == IA_CSS_PIPE_ID_CAPTURE) {
 #if defined (IS_ISP_2500_SYSTEM)
 		ref_info               = mycs_capture->primary_binary[0].internal_frame_info;
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_firmware.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_firmware.c
index 3268e29..27659e8 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_firmware.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_firmware.c
@@ -47,10 +47,10 @@ struct fw_param {
 /* Warning: same order as SH_CSS_BINARY_ID_* */
 static struct firmware_header *firmware_header;
 
-/* The string STR(irci_master_20150129_1500) is a place holder
+/* The string STR(irci_master_20150131_0457) is a place holder
  * which will be replaced with the actual RELEASE_VERSION
  * during package generation. Please do not modify  */
-static const char *release_version = STR(irci_master_20150129_1500);
+static const char *release_version = STR(irci_master_20150131_0457);
 
 #define MAX_FW_REL_VER_NAME	300
 static char FW_rel_ver_name[MAX_FW_REL_VER_NAME] = "---";
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_params.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_params.c
index 4617c6a..5f8e9fa 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_params.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_params.c
@@ -2860,9 +2860,10 @@ ia_css_isp_3a_statistics_allocate(const struct ia_css_3a_grid_info *grid)
 
 	IA_CSS_ENTER("grid=%p", grid);
 
-/* MW: Does "grid->enable" also control the histogram output ?? */
-#if !defined(IS_ISP_2500_SYSTEM)
 	assert(grid != NULL);
+
+	/* MW: Does "grid->enable" also control the histogram output ?? */
+#if !defined(IS_ISP_2500_SYSTEM)
 	if (!grid->enable)
 		return NULL;
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/spmem_dump.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/spmem_dump.c
index 693914e..65e7e98 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/spmem_dump.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/spmem_dump.c
@@ -24,7 +24,7 @@
 
 /* function input_system_acquisition_stop: ADA */
 
-/* function longjmp: 64C5 */
+/* function longjmp: 64C6 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_HIVE_IF_SRST_MASK
@@ -66,7 +66,7 @@
 
 /* function ia_css_queue_load: 4AF3 */
 
-/* function setjmp: 64CE */
+/* function setjmp: 64CF */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_sp2host_isys_event_queue
@@ -80,7 +80,7 @@
 #define HIVE_ADDR_sp_sem_for_sp2host_isys_event_queue 0x4640
 #define HIVE_SIZE_sp_sem_for_sp2host_isys_event_queue 20
 
-/* function ia_css_dmaproxy_sp_wait_for_ack: 69FC */
+/* function ia_css_dmaproxy_sp_wait_for_ack: 69FD */
 
 /* function ia_css_sp_rawcopy_func: 4D18 */
 
@@ -370,7 +370,7 @@
 
 /* function ia_css_pipeline_sp_run: 13DF */
 
-/* function memcpy: 656E */
+/* function memcpy: 656F */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_GP_DEVICE_BASE
@@ -404,7 +404,7 @@
 
 /* function ia_css_circbuf_increase_size: E3C */
 
-/* function __divu: 64EC */
+/* function __divu: 64ED */
 
 /* function ia_css_thread_sp_get_state: C7F */
 
@@ -438,7 +438,7 @@
 
 /* function ia_css_spctrl_sp_set_state: 5614 */
 
-/* function ia_css_thread_sem_sp_signal: 6733 */
+/* function ia_css_thread_sem_sp_signal: 6734 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_IRQ_BASE
@@ -464,13 +464,13 @@
 #define HIVE_ADDR_sp_TIMED_CTRL_BASE 0x40
 #define HIVE_SIZE_sp_TIMED_CTRL_BASE 4
 
-/* function ia_css_isys_sp_isr: 6BAD */
+/* function ia_css_isys_sp_isr: 6BAE */
 
 /* function ia_css_isys_sp_generate_exp_id: 5DB6 */
 
 /* function ia_css_rmgr_sp_init: 5EA0 */
 
-/* function ia_css_thread_sem_sp_init: 6804 */
+/* function ia_css_thread_sem_sp_init: 6805 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_is_isp_requested
@@ -520,7 +520,7 @@
 
 /* function ia_css_spctrl_sp_get_spid: 561B */
 
-/* function ia_css_dmaproxy_sp_read_byte_addr: 6A2A */
+/* function ia_css_dmaproxy_sp_read_byte_addr: 6A2B */
 
 /* function ia_css_rmgr_sp_uninit: 5E99 */
 
@@ -696,7 +696,7 @@
 #define HIVE_ADDR_sp_host2sp_isys_event_queue_handle 0x4B98
 #define HIVE_SIZE_sp_host2sp_isys_event_queue_handle 12
 
-/* function ia_css_thread_sp_yield: 66AC */
+/* function ia_css_thread_sp_yield: 66AD */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_all_cbs_param_desc
@@ -854,7 +854,7 @@
 
 /* function ia_css_rmgr_sp_refcount_init_vbuf: 5F3B */
 
-/* function dmaproxy_sp_read_write: 6AB5 */
+/* function dmaproxy_sp_read_write: 6AB6 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_dmaproxy_isp_dma_cmd_buffer
@@ -892,7 +892,7 @@
 #define HIVE_ADDR_sp_ia_css_flash_sp_in_service 0x3178
 #define HIVE_SIZE_sp_ia_css_flash_sp_in_service 4
 
-/* function ia_css_dmaproxy_sp_process: 682C */
+/* function ia_css_dmaproxy_sp_process: 682D */
 
 /* function ia_css_tagger_buf_sp_mark_from_end: 294D */
 
@@ -1136,7 +1136,7 @@
 
 /* function cnd_input_system_cfg: 392 */
 
-/* function __ia_css_sp_rawcopy_func_critical: 6B36 */
+/* function __ia_css_sp_rawcopy_func_critical: 6B37 */
 
 /* function ia_css_dmaproxy_sp_set_width_exception: 2FF0 */
 
@@ -1146,7 +1146,7 @@
 
 /* function ia_css_tagger_buf_sp_pop_unmarked_and_unlocked: 26EC */
 
-/* function __modu: 6532 */
+/* function __modu: 6533 */
 
 /* function ia_css_dmaproxy_sp_init_isp_vector: 2E75 */
 
@@ -1166,7 +1166,7 @@
 
 /* function ia_css_queue_local_init: 481B */
 
-/* function sp_event_proxy_callout_func: 65C4 */
+/* function sp_event_proxy_callout_func: 65C5 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_thread_sp_num_ready_threads
@@ -1198,7 +1198,7 @@
 
 /* function ia_css_queue_dequeue: 4699 */
 
-/* function ia_css_dmaproxy_sp_configure_channel: 6A41 */
+/* function ia_css_dmaproxy_sp_configure_channel: 6A42 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_current_thread_fiber_sp
@@ -1468,17 +1468,17 @@
 #define HIVE_ADDR_sp_sp_isp_input_stream_format 0x40C8
 #define HIVE_SIZE_sp_sp_isp_input_stream_format 20
 
-/* function __mod: 651E */
+/* function __mod: 651F */
 
 /* function ia_css_dmaproxy_sp_init_dmem_channel: 2F33 */
 
 /* function ia_css_thread_sp_join: CFB */
 
-/* function ia_css_dmaproxy_sp_add_command: 6B20 */
+/* function ia_css_dmaproxy_sp_add_command: 6B21 */
 
 /* function ia_css_sp_metadata_thread_func: 54C1 */
 
-/* function __sp_event_proxy_func_critical: 65B1 */
+/* function __sp_event_proxy_func_critical: 65B2 */
 
 /* function ia_css_sp_metadata_wait: 55D4 */
 
@@ -1528,7 +1528,7 @@
 #define HIVE_ADDR_sp_ia_css_isys_sp_frontend_states 0x6270
 #define HIVE_SIZE_sp_ia_css_isys_sp_frontend_states 12
 
-/* function ia_css_dmaproxy_sp_read_byte_addr_mmio: 6A13 */
+/* function ia_css_dmaproxy_sp_read_byte_addr_mmio: 6A14 */
 
 /* function ia_css_ispctrl_sp_done_ds: 3469 */
 
@@ -1842,7 +1842,7 @@
 
 /* function ia_css_sp_isp_param_mem_load: 4304 */
 
-/* function __div: 64D6 */
+/* function __div: 64D7 */
 
 /* function ia_css_isys_sp_frontend_create: 5AC1 */
 
@@ -1860,7 +1860,7 @@
 #define HIVE_ADDR_sp_ia_css_flash_sp_in_use 0x4A58
 #define HIVE_SIZE_sp_ia_css_flash_sp_in_use 4
 
-/* function ia_css_thread_sem_sp_wait: 677E */
+/* function ia_css_thread_sem_sp_wait: 677F */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_sleep_mode
-- 
1.9.1

